mobile/packages/swap/adapters/api/minswap/transformers.ts

```diff
@@ -0,0 +1,364 @@
+import {isNonNullable} from '@yoroi/common'
+import {Portfolio, Swap} from '@yoroi/types'
+
+import {freeze} from 'immer'
+
+import {
+  CancelResponse,
+  CreateRequest,
+  CreateResponse,
+  Dex,
+  EstimateRequest,
+  EstimateResponse,
+  LimitOptionsResponse,
+  MinswapApiConfig,
+  PendingOrdersResponse,
+  TokensRequest,
+  TokensResponse,
+} from './types'
+
+const mapProtocolToDex = (protocol: Swap.Protocol): Dex => {
+  switch (protocol) {
+    case Swap.Protocol.Minswap_v2:
+      return Dex.MinswapV2
+    case Swap.Protocol.Minswap_v1:
+      return Dex.Minswap
+    case Swap.Protocol.Minswap_stable:
+      return Dex.MinswapStable
+    case Swap.Protocol.Muesliswap:
+      return Dex.MuesliSwap
+    case Swap.Protocol.Splash_v1:
+      return Dex.Splash
+    case Swap.Protocol.Sundaeswap_v3:
+      return Dex.SundaeSwapV3
+    case Swap.Protocol.Sundaeswap_v1:
+      return Dex.SundaeSwap
+    case Swap.Protocol.Vyfi_v1:
+      return Dex.VyFinance
+    case Swap.Protocol.Cswap:
+      return Dex.CswapV1
+    case Swap.Protocol.Wingriders_v2:
+      return Dex.WingRidersV2
+    case Swap.Protocol.Wingriders_v1:
+      return Dex.WingRiders
+    case Swap.Protocol.Wingriders_stable:
+      return Dex.WingRidersStableV2
+    case Swap.Protocol.Spectrum_v1:
+      return Dex.Spectrum
+    case Swap.Protocol.Splash_v1:
+      return Dex.SplashStable
+    default:
+      return Dex.Unsupported
+  }
+}
+
+export const transformersMaker = (config: MinswapApiConfig) => {
+  const {isPrimaryToken, primaryTokenInfo, address, partner} = config
+
+  // Convert portfolio token ID to API token ID (like DexHunter)
+  const toTokenId = (tokenId: Portfolio.Token.Id) => {
+    const result = isPrimaryToken(tokenId)
+      ? 'lovelace'
+      : tokenId.replace('.', '')
+    return result
+  }
+
+  const transformToken = (token: {
+    token_id: string
+    logo: string | null
+    ticker: string | null
+    is_verified: boolean | null
+    price_by_ada: number | null
+    project_name: string | null
+    decimals: number | null
+  }): Portfolio.Token.Info => {
+    const formattedTokenId: Portfolio.Token.Id =
+      token.token_id === 'lovelace'
+        ? primaryTokenInfo.id
+        : `${token.token_id.slice(0, 56)}.${token.token_id.slice(56)}`
+
+    return freeze(
+      {
+        id: formattedTokenId,
+        name: token.project_name || token.ticker || 'Unknown Token',
+        ticker: token.ticker || '',
+        decimals: token.decimals || 6,
```

Comment:
fallback is always 0 for decimals
```suggestion
        decimals: token.decimals ?? 0
```